File: LongestPalindromicSubstring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        if(len(s) <= 1):
            return s
        n = len(s)
        max_len = 1
        start = 0
        end = 0
        for i in range(n-1):
            left = i
            right = i
            while(left >= 0 and right < n):
                if(s[left] == s[right]):
                    left = left-1
                    right = right+1
                else:
                    break
            _len = right-left-1
            if(_len > max_len):
                max_len = _len
                start = left+1
                end = right-1
        for i in range(n-1):
            left = i
            right = i+1
            while(left >= 0 and right < n):
                if(s[left] == s[right]):
                    left = left-1
                    right = right+1
                else:
                    break
            _len = right-left-1
            if(_len > max_len):
                max_len = _len
                start = left+1
                end = right-1

        return s[start:end+1]
            
        
            
        

File: Longest Palindromic Substring.java
class Solution {
    public boolean check_palindrome(String str){
        int start = 0;
        int end = str.length()-1;
        while(start < end){
            if(str.charAt(start) != str.charAt(end)){
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
    public String longestPalindrome(String s) {
        String result = "";
        for(int i=0;i<s.length();i++){
            for(int j=i+1;j<=s.length();j++){
                String temp = s.substring(i,j);
                if(check_palindrome(temp)){
                    if(temp.length() > result.length()){
                        result = temp;
                    }
                }
            }
    }
    return result;
    }
}
